src/test/java/seedu/address/model/EventBookTest.java

```diff
@@ -0,0 +1,118 @@
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalEvents.BINGO;
+import static seedu.address.testutil.TypicalEvents.getTypicalEventBook;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.event.Event;
+import seedu.address.model.event.exceptions.DuplicateEventException;
+import seedu.address.testutil.EventBuilder;
+
+public class EventBookTest {
+
+    private final EventBook eventBook = new EventBook();
+
+    @Test
+    public void constructor() {
+        assertEquals(Collections.emptyList(), eventBook.getEventList());
+    }
+
+    @Test
+    public void resetData_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> eventBook.resetData(null));
+    }
+
+    @Test
+    public void resetData_withValidReadOnlyEventBook_replacesData() {
+        EventBook newData = getTypicalEventBook();
+        eventBook.resetData(newData);
+        assertEquals(newData, eventBook);
+    }
+
+    @Test
+    public void resetData_withDuplicateEvents_throwsDuplicateEventException() {
+        // Two events with the same identity fields
+        Event editedBingo = new EventBuilder(BINGO)
+                .build();
+        List<Event> newEvents = Arrays.asList(BINGO, editedBingo);
+        EventBookStub newData = new EventBookStub(newEvents);
+
+        assertThrows(DuplicateEventException.class, () -> eventBook.resetData(newData));
+    }
+
+    @Test
+    public void hasEvent_nullEvent_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> eventBook.hasEvent(null));
+    }
+
+    @Test
+    public void hasEvent_eventNotInEventBook_returnsFalse() {
+        assertFalse(eventBook.hasEvent(BINGO));
+    }
+
+    @Test
+    public void hasEvent_eventInEventBook_returnsTrue() {
+        eventBook.addEvent(BINGO);
+        assertTrue(eventBook.hasEvent(BINGO));
+    }
+
+    @Test
+    public void hasEvent_eventWithSameIdentityFieldsInEventBook_returnsTrue() {
+        eventBook.addEvent(BINGO);
+        Event editedBingo = new EventBuilder(BINGO)
+                .build();
+        assertTrue(eventBook.hasEvent(editedBingo));
+    }
+
+    @Test
+    public void getEventList_modifyList_throwsUnsupportedOperationException() {
+        assertThrows(UnsupportedOperationException.class, () -> eventBook.getEventList().remove(0));
+    }
+
+    @Test
+    public void equals_sameObject_returnsTrue() {
+        EventBook eventBook = new EventBook();
+        assertTrue(eventBook.equals(eventBook));
+    }
+
+    @Test
+    public void equals_differentClass_returnsFalse() {
```

Comment:
Perhaps name it as `differentObject` for consistency? not necessary tho just cosmetic change.